src/middleware/visualizer.coffee

```diff
@@ -7,11 +7,43 @@ config.visualizer = config.get('visualizer')
 
 minEvPeriod = config.visualizer.minimumEventPeriodMillis ? 100
 
+enableTSNormalization = config.visualizer.enableTSNormalization ? false
+orchestrationTSBufferMillis = config.visualizer.orchestrationTSBufferMillis ? 100
+
 formatTS = (ts) -> moment(new Date(ts)).valueOf()
 
-addRouteEvents = (dst, route, prefix) ->
+# function to get the TimeStamp difference
+getTSDiff = ( CTXStartTS, obj ) ->
+  # default TS
+  earliestTS = 0
+
+  # foreach record in object
+  for record in obj
+    # get request timestamp
+    ts = formatTS record.request.timestamp
+
+    # if TS earlier then update
+    if earliestTS < ts then earliestTS = ts
+
+  # CTXStartTS minus earlistTS to get TS diff
+  TSDiff = CTXStartTS - earliestTS
```

Comment:
Style: rather use lower case for variable names, e.g. `tsDiff`, even though its an abbreviation. However I think something like `earliestTS` is fine though.